main.rb

```diff
@@ -0,0 +1,141 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+# Module Enumerable
+module Enumerable
+  def my_each
+    i = 0
+    return enum_for unless block_given?
+
+    arr = to_a
+    while i < arr.size
+      yield(arr.at(i))
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    return enum_for unless block_given?
+
+    arr = to_a if is_a?(Range) || is_a?(Hash)
+    while i < arr.size
+      yield(arr.at(i), i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    selected = []
+    my_each { |item| selected << item if yield(item) }
+    selected
+  end
+
+  def my_all?(args = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+    elsif args.nil?
+      my_each { |item| return false if item.nil? || item == false }
+    elsif !args.nil? && (args.is_a? Class)
+      my_each { |item| return false if item.class != args }
+    elsif !args.nil? && instance_of?(Hash)
+      return false unless empty?
+    elsif !args.nil? && args.instance_of?(Regexp)
+      my_each { |item| return false unless args.match(item.to_s) }
+    else
+      my_each { |item| return false if item != args }
+    end
+    true
+  end
```

Comment:
- [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric.
This test should return true.
```
  arr = [1,2,3,4,5]
  p arr.my_all?(Numeric)

```